django/chat/responses.py

```diff
@@ -31,12 +33,17 @@
 @permission_required("chat.access_message", objectgetter(Message, "message_id"))
 def otto_response(request, message_id=None):
     """
-    Stream a response to the user's message. Uses LangChain to manage chat history.
+    Stream a response to the user's message. Uses LlamaIndex to manage chat history.
     """
     response_message = Message.objects.get(id=message_id)
     chat = response_message.chat
     assert chat.user_id == request.user.id
     mode = chat.options.mode
+
+    # For costing and logging. Contextvars are accessible anytime during the request
+    # including in async functions (i.e. htmx_stream) and Celery tasks.
+    bind_contextvars(message_id=message_id, feature=mode)
```

Comment:
By adding to the structlog contextvars, we can use this information in the Cost objects whenever they are created (including from async functions or from Celery tasks)